### PR TITLE
Deployment tweaks for ubuntu on ot55

### PR DESCRIPTION
**Pull request by jimallman**
NB this is failing when trying to install peyotl dependencies:
```
TASK [peyotl : Install peyotl python requirements] ************************************************************************************************************
task path: /Users/jima/projects/nescent/opentree/otindex_ansible/roles/peyotl/tasks/main.yml:24
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ImportError: No module named pkg_resources
fatal: [devotindex]: FAILED! => {"changed": false, "msg": "Failed to import the required Python library (setuptools) on ip-172-30-0-168's Python /usr/bin/python. Please read the module documentation and install it in the appropriate location. If the required library is installed, but Ansible is using the wrong Python
Deployment tweaks for ubuntu on ot55
 interpreter, please consult the documentation on ansible_python_interpreter"}
```
I believe we're assuming prior package installations that were done in a different python environment. There are at least two `virtualenv`s on the server:
 - `/home/opentree/repo/../venv/`, running python2.7
 - `/home/opentree/repo/otindex/venv-otindex/`, running python3.6

The first one is documented in the Ansible playbook as `{{install_dir}}/venv`. I'm not sure who created the second one, but it was created on the same day as the previous deployment. @kcranston @snacktavish Thoughts?

### CODE REVIEW

**Comment by jimallman** [comment]
Possible clues (and solutions?) re Ansible's confusion over python interpreters:
https://docs.ansible.com/ansible/latest/reference_appendices/interpreter_discovery.html

File: hosts [hunk 1]
```diff
@@ -7,11 +7,13 @@
...
-otindexdev ansible_host=otindexdev.opentreeoflife.org ansible_user=admin ansible_ssh_private_key_file=/home/ejmctavish/.ssh/opentree.pem
-
+##devotindex ansible_host=devotindex.opentreeoflife.org ansible_user=admin ansible_ssh_private_key_file=/Users/jima/.ssh/id_rsa
+##devotindex ansible_host=devotindex.opentreeoflife.org ansible_user=admin ansible_ssh_private_key_file=/Users/jima/.ssh/opentree/production.pem
+devotindex ansible_host=devotindex.opentreeoflife.org ansible_user=ubuntu ansible_ssh_private_key_file=/Users/jima/.ssh/opentree/production.pem
 
 [devapi]
-devapi ansible_host=devapi.opentreeoflife.org ansible_user=admin ansible_ssh_private_key_file=/home/ejmctavish/.ssh/opentree.pem
+#devapi ansible_host=devapi.opentreeoflife.org ansible_user=admin ansible_ssh_private_key_file=/Users/jima/.ssh/id_rsa
+devapi ansible_host=devapi.opentreeoflife.org ansible_user=admin ansible_ssh_private_key_file=/Users/jima/.ssh/opentree/production.pem
```
Comment (kcranston) [inline hosts:16]: why are we putting the key in a specific user (jima, ejmctavish) dir rather than a generic user?

**Comment by kcranston** [comment]
Looks like the second virtualenv was created manually, which is not ideal. Note that the pip role can take a specific python version, if we want to be explicit, i.e. 

```
- name: Install peyotl python requirements
  pip:
    requirements: "{{install_dir}}/peyotl/requirements.txt"
    virtualenv: "{{install_dir}}/../venv"
    virtualenv_python: "{{ python_version }}"
```

File: hosts (hunk 1 above)
Comment (snacktavish) [reply hosts:16]: I think this was changed as as a marker that we weren't using a shared key anymore, and were each using our own?

**Comment by jimallman** [comment]
> Looks like the second virtualenv was created manually, which is not ideal. 

Yeah, I'm inclined to clobber this venv entirely and see what happens. 

> Note that the pip role can take a specific python version, if we want to be explicit,

I'll slow down and look at each task. Clearly we're installing packages in one env and trying to use them in another. Just wanted to make sure I wasn't making a dumb mistake.

**Comment by snacktavish** [comment]
Hmm yes - I suspect the double virtual envs may have been due to needing different python versions for different dependencies (That is how the otcetera install was (is?) set up), or conflicting install procedures, orrr maybe just cruft form a first failed install that didn't get wiped. But I agree with clobbering!

File: hosts (hunk 1 above)
Comment (jimallman) [reply hosts:16]: I'm happy to try and formalize an approach, if we want to go back to shared keys. Honestly, I just kept trying different auth methods until I found a user+key that would connect to ot55. 😬

**Comment by kcranston** [comment]
Not sure what you were intending, but that [most recent commit](https://github.com/OpenTreeOfLife/otindex_ansible/pull/6/commits/98556e8adba4a73297553f27e23a490f743d1446) sets the python version for running ansible on the host, not for the virtualenv.

**Comment by jimallman** [comment]
That makes sense. I'm still trying to wrap my head around this. This change was the only way I could make [this task](https://github.com/OpenTreeOfLife/otindex_ansible/blob/fad8c02d5e05f0f8fffa162ef20564b4cf18a988/roles/peyotl/tasks/main.yml#L24-L29) complete successfully. I've tried to add `virtualenv_python` instead [as you suggested above](https://github.com/OpenTreeOfLife/otindex_ansible/pull/6#issuecomment-887612674), but I'm still getting weird error messages that imply we're installing to the system python `/usr/bin/python`.

This [old StackOverflow comment](https://stackoverflow.com/questions/26402123/ansible-creating-a-virtualenv#comment69325723_30777607) describes the symptoms I'm seeing, but without a clear explanation for it.

I'm backing out the global `ansible_python_interpreter` in case you want to try this for yourself.

**Comment by jimallman** [comment]
NB - Latest commits fix the vexing warning about missing setuptools when installing peyotl dependencies.

**Comment by jimallman** [comment]
I've made a few  more changes to support deployment to modern (2022) Ubuntu. Working now on **devotindex**!

**Pull request merged**